### PR TITLE
Create library and add install

### PR DESCRIPTION
**Pull request by streeve**
Also allow CXX standard newer than 14

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+file(GLOB HACCABANA_HEADERS GLOB *.h)
```
Comment (junghans) [inline src/CMakeLists.txt:1]:
```suggestion
file(GLOB HACCABANA_HEADERS *.h)
```

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+add_library(HACCabana ${HACCABANA_SOURCES})
+
+target_include_directories(HACCabana PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}
+                           $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+                           $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
+
+target_link_libraries(HACCabana Cabana::cabanacore)
+
+install(TARGETS HACCabana DESTINATION lib)
```
Comment (junghans) [inline src/CMakeLists.txt:14]:
```suggestion
install(TARGETS HACCabana DESTINATION ${CMAKE_INSTALL_LIBDIR})
```

**Review by junghans — CHANGES_REQUESTED** [review]
Maybe we should add an export as well.

**Comment by streeve** [comment]
@junghans ready for review any time

**Review by junghans — APPROVED** [review]

**Comment by streeve** [comment]
@steverangel @adrianpope are you guys good with these changes? Should just make the build a little more robust

**Comment by steverangel** [comment]
I think this all looks fine.

**Pull request merged**